MetovaTestKit/AsynchronousTesting/XCTestCase+AsyncTesting.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  XCTestCase+AsyncTesting.swift
+//  MetovaTestKit
+//
+//  Created by Logan Gauthier on 5/5/17.
+//  Copyright © 2017 Metova. All rights reserved.
+//
+//  MIT License
+//
+//  Permission is hereby granted, free of charge, to any person obtaining
+//  a copy of this software and associated documentation files (the
+//  "Software"), to deal in the Software without restriction, including
+//  without limitation the rights to use, copy, modify, merge, publish,
+//  distribute, sublicense, and/or sell copies of the Software, and to
+//  permit persons to whom the Software is furnished to do so, subject to
+//  the following conditions:
+//
+//  The above copyright notice and this permission notice shall be
+//  included in all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+//  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+//  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+//  LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+//  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+//  WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import XCTest
+
+extension XCTestCase {
+    
+    /// Provides a succinct syntax for performing a simple asynchronous test.
+    ///
+    /// - Parameters:
+    ///   - timeInterval: The number of seconds to wait before running `testAction`.
+    ///   - testAction: The closure which will have a delayed execution.
+    ///
+    /// - Warning: This method can only be used once within a single test method as it uses `waitForExpectations(timeout:handler:)`.
+    public func MTKPerformAsyncTest(after timeInterval: TimeInterval, testAction: @escaping () -> Void) {
+        
+        let waitExpectation = expectation(description: "Waiting to resume tests.")
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + timeInterval) {
+            testAction()
+            waitExpectation.fulfill()
+        }
+        
+        waitForExpectations(timeout: timeInterval + 5, handler: nil)
```

Comment:
Actually, I suppose the tricky part here is that we have no idea how long the user expects their block of code to run.  If, for example, they put some long-running process here, if it took more than 5 seconds to complete execution, then this test would fail, and maybe that's not expected.

However... is there any reason that we really need to timeout the wait?  We should just pass some really large value.  The wait will return pretty immediately after all the expectations are fulfilled.  Therefore, if this method hangs, it can only be because the `testAction` block never returned, allowing us to fulfill the expectation.